### PR TITLE
add travis support for basic linting

### PR DESCRIPTION
**Pull request by stevemar**
There were at least two syntax errors that should **not** have existed in a published pattern. A simple CI job that performs basic linting is usually a requirement for a code pattern. 

This PR fixes the aforementioned syntax errors, adds travis support to call a linting job (flake8) and modifies the code base to be flake8 compliant (ignoring long lines), and py3 friendly

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+language: python
+python:
+  - "2.7"
+  - "3.5"
+install:
+  - "pip install flake8"
+script:
+ - flake8 . --ignore E501,F841
```
Comment (stevemar) [inline .travis.yml]: this will ignore long lines and unused variables see line 112 of record_audio.py

File: audio/detect_hotword.py [hunk 2]
```diff
@@ -1,10 +1,14 @@
-import snowboydecoder
 import sys
 import signal
+
+import snowboydecoder
```
Comment (stevemar) [inline audio/detect_hotword.py:4]: imports are organized as follow:

1. built into python
2. 3rd party
3. same project

with new lines separating them

File: audio/record_audio.py [hunk 3]
```diff
@@ -1,14 +1,9 @@
-import pyaudio
-import httplib
-import wave
 import audioop
 from collections import deque
-import os
-import requests
 import math
-from os.path import join, dirname
+import wave
```
Comment (stevemar) [inline audio/record_audio.py:4]: this had a bunch of unused imports

File: audio/record_audio.py [hunk 4]
```diff
@@ -37,7 +36,7 @@ def audio_int(num_samples=50):
         is the avg of the 20% largest intensities recorded.
     """
 
-    print "Getting intensity values from mic."
+    print("Getting intensity values from mic.")
```
Comment (stevemar) [inline audio/record_audio.py:39]: print statements need to be wrapped in `(` to be py3 friendly

File: ibm_cloud_functions/iot-pub.py [hunk 5]
```diff
@@ -1,10 +1,11 @@
...
+
+
 def main(dict):
     iot_org_id = dict['iot_org_id']
     device_id = dict['device_id']
     device_type = dict['device_type']
     api_token = dict['api_token']
     classes = [nlc_class for nlc_class in dict['msg']['classes'] if nlc_class['confidence'] > 0.1]
-    r = requests.post('http://' + iot_org_id + '.messaging.internetofthings.ibmcloud.com:1883/api/v0002/device/types/' + device_type '/devices/' + device_id + '/events/query', headers={'Content-Type': 'application/json'}, json=classes, auth=('use-token-auth', api_token))
+    requests.post('http://' + iot_org_id + '.messaging.internetofthings.ibmcloud.com:1883/api/v0002/device/types/' + device_type + '/devices/' + device_id + '/events/query', headers={'Content-Type': 'application/json'}, json=classes, auth=('use-token-auth', api_token))
```
Comment (stevemar) [inline ibm_cloud_functions/iot-pub.py]: fixed syntax error here

File: audio/record_audio.py [hunk 6]
```diff
@@ -75,44 +73,44 @@ def listen_for_speech(threshold=THRESHOLD, num_phrases=1):
...
-    print "* Done recording"
+    print("* Done recording")
     stream.close()
     p.terminate()
 
+
 def transcribe_audio(path_to_audio_file):
     encoded_audio = open('speech.wav', 'r').read().encode("base64")
-    // TODO: use watson speech to text service
+    # TODO: use watson speech to text service
```
Comment (stevemar) [inline audio/record_audio.py:113]: fixed syntax error here, comments are written with a `#`, this ain't java.

**Review by cclauss — APPROVED** [review]
Nice!

File: ibm_cloud_functions/iot-pub.py (hunk 5 above)
Comment (cclauss) [reply ibm_cloud_functions/iot-pub.py]: __dict__ is a poor choice for a variable name because it shadows the builtin __dict()__.

Comment (stevemar) [reply ibm_cloud_functions/iot-pub.py]: agreed, i'll change it.

File: .travis.yml (hunk 1 above)
Comment (cclauss) [reply .travis.yml]: Python 3.5?  Why not jump to 3.6?  (3.7 is in beta)

Comment (stevemar) [reply .travis.yml]: good point. i copied an old travis file. will update to 3.6, shouldn't require any code changes

**Review by cclauss — APPROVED** [review]

**Review by scottdangelo — APPROVED** [review]

**Comment by stevemar** [comment]
thanks @cclauss and @scottdangelo for the reviews

**Pull request merged**